specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchindex.json

```diff
@@ -1455,7 +1455,7 @@
         "lexicon"
       ],
       "x-ms-enum": {
-        "name": "Speller",
+        "name": "QuerySpellerType",
```

Comment:
Does `Type` add anything to this name?  Most enums are better without an extra `Type` at the end.

Reply:
You are right - in our discussion we only decided on `QueryAnswerType` rename for the `Answers` parameter.
We have 3 params that are similar in nature and usage and it would be good to keep them consistently named - `Speller`->`QuerySpellerType`, `Answers`->`QueryAnswerType` and `Captions`->`QueryCaptionType` - all extensible enums.

One pro of adding the suffix `Type` is that the param can otherwise be misunderstood as containing part of a result object. It is best illustrated for `QueryAnswer` and `QueryCaption`. We could leave the third as `QuerySpeller` or tack on `Type` there too to make all 3 names closely grouped.

Reply:
Also, I made additional renames (Removing all instances of `SearchIndexer` prefix from the `KnowledgeStore.*` types) here that were earlier done during the .NET SDK API review.
Please take a look.

Reply:
@tg-msft - what do you think?
Also, do you know how should I go forward with this PR with the validation errors regarding changes in an existing swagger? Do I need to send an email, or can I tag folks here?